src/main/java/com/mathworks/ci/MatlabRunTestsStepExecution.java

```diff
@@ -68,12 +68,16 @@ private int execMatlabCommand(FilePath workspace, Launcher launcher,
                     "addpath('" + genScriptLocation.getRemote().replaceAll("'", "''") + "'); ";
             final String matlabScriptName = getValidMatlabFileName(genScriptLocation.getBaseName());
 
-            ProcStarter matlabLauncher = getProcessToRunMatlabCommand(workspace, launcher, listener,
-                    envVars, cmdPrefix + matlabScriptName, uniqueTmpFldrName);
+            ProcStarter matlabLauncher =
+                    getProcessToRunMatlabCommand(workspace, launcher, listener, envVars,
+                            cmdPrefix + matlabScriptName + ",delete('.matlab/"
+                                    + genScriptLocation.getBaseName() + "/" + matlabScriptName
+                                    + ".m'),run(runnerScript),rmdir(tmpDir,'s')",
```

Comment:
I thought that last week we ran into a problem with the run function adding a problematic cd. Why isn't that a problem here?

Reply:
Remember now, ts because earlier the temp file was directly written into /tmp instead /tmp/random_folder. hence it was trying to CD some  folder in /tmp and running into issue there. for safer side i have removed  run () anyways